docs - update sync, scan troubleshooting (#26572)
metabase/metabase: docs/databases/connecting.md

## Syncing and scanning databases

> Metabase runs sync and scan queries in order to show tables, views, and columns, populate dropdown menus with the right values, and suggest helpful visualizations. Metabase does _not_ copy any data from your database---it only maintains lists of the tables and columns.

Reviewer: I would argue that populating dropdown lists _is_ copying data.

![Re-scan options](./images/re-scan-options.png)

> ### Syncing and scanning using the API

Author: I moved this from the Troubleshooting guide. Do we want to keep API info in the reference doc here?

Reviewer 2: I think it's okay here, as this topic is sufficiently advanced, and we don't go into too much detail. If more detail is required, we should include that in the endpoint docs themselves.

> Metabase syncs and scans regularly, but if the database administrator has just changed the database schema, or if a lot of data is added automatically at specific times, you may want to write a script that uses the [Metabase API](https://www.metabase.com/learn/administration/metabase-api) to force sync or scan to take place right away. [Our API](../api-documentation.md) provides two ways to do this:

Reviewer 2: ```suggestion
Metabase syncs and scans regularly, but if the database administrator has just changed the database schema, or if a lot of data is added automatically at specific times, you may want to write a script that uses the [Metabase API](https://www.metabase.com/learn/administration/metabase-api) to force a sync or scan. [Our API](../api-documentation.md) provides two ways to initiate a sync or scan of a database:
```

> 1. Using an endpoint with a session token: `/api/database/:id/sync_schema` or `api/database/:id/rescan_values`. These do the same things as going to the database in the Admin Panel and choosing **Sync database schema now** or **Re-scan field values now** respectively. In this case you have to authenticate with a user ID and pass a session token in the header of your request.
> 
> 2. Using an endpoint with an API key: `/api/notify/db/:id`. This endpoint was made to notify Metabase to sync after an [ETL operation](https://www.metabase.com/learn/analytics/etl-landscape) finishes. In this case you must pass an API key by defining the `MB_API_KEY` environment variable.

Reviewer 2: ```suggestion
1. Using a a session token:  the `/api/database/:id/sync_schema` or `api/database/:id/rescan_values` endpoints. These endpoints do the same things as going to the database in the Admin Panel and choosing **Sync database schema now** or **Re-scan field values now** respectively. To use these endpoints, you have to authenticate with a user ID and pass a session token in the header of your request.

2. Using an API key: `/api/notify/db/:id`. We created this endpoint so that people could notify their Metabase to sync after an [ETL operation](https://www.metabase.com/learn/analytics/etl-landscape) finishes. To use this endpoint, you must pass an API key by defining the `MB_API_KEY` environment variable.
```
